python/cac_tripplanner/templates/service-worker.js

```diff
@@ -28,7 +28,7 @@ self.addEventListener('fetch', function(event) {
         } else {
             return fetch(event.request).then(function (response) {
                 // cache fetched request if it is on this domain
-                if (request.url.startsWith(location.href)) {
+                if (!location.href.includes('/admin') && request.url.startsWith(location.href)) {
```

Comment:
This is a troublesome one.

I've finally gotten it to stop doing the crash.  I think the fact that I had earlier installed the site as an app meant that no matter how much I tried to get rid of the service worker (by either "Clear site data" under Application or in chrome://serviceworker-internals/), it didn't delete the one saved with the app, and just kept using it.  To make matters more confusing, when I clicked on `service-worker.js` in the Application tab
![image](https://user-images.githubusercontent.com/6598836/30123610-9872e330-9300-11e7-9c63-d6b8c729be71.png)
it didn't open in Sources, it opened a new tab with http://localhost:8024/service-worker.js, which of course showed the version being served, which was not the version actually being used by my browser.

Uninstalling GoPhillyGo from my Apps and restarting my browser finally cleared that stale service worker, which means both that I can actually see changes and that clicking it in the Application tab opens it in Sources where I can drop a debugger.

I also found the actual storage location of the `cac_tripplanner_v2` cache (under `~/.config/google-chrome/Default/Service Worker/CacheStorage`) and deleted it, but that might have been a red herring.

Anyway, now that I can get the new service worker and don't have stale stuff in my cache, I can confirm that the crashes go away.

However, I think there are some problems with this check in both its current and former forms.  

When I drop a debugger at this line and run the test in the console, I get "Uncaught ReferenceError: request is not defined".  I think that's getting caught by the `catch`, but it means we're not actually putting anything in the cache.  If I change it to `event.request.url`, it still always comes up false because it appears that `location.href`, in the context of the service worker, is always http://localhost:8024/service-worker.js.

So I think the test should be something like `!event.request.url.includes('/admin') && event.request.url.startsWith(location.origin)` to exclude 3rd party and admin stuff but cache everything else on the domain.

Then again, if the existing condition was no good, there shouldn't have been anything in the service worker's cache at all, so how could we have seen an error caused by over-caching?  I may still be missing something.

Reply:
I suspect the debugger might be untrustworthy in this context, as I've also noticed console logs to not print as expected from the service worker. The URL checks had come from some documentation, though I'll have to dig some to find the reference. Given that the caching had been occurring and now no longer cache admin files, I think that should be enough to go on along with behavior to review.